Tests/Core/BasicTests.swift

```diff
@@ -62,7 +62,10 @@ class BasicTests: OperationTests {
         var finalCompletionBlockDidRun = 0
         operation.addCompletionBlock {
             finalCompletionBlockDidRun += 1
-            expectation.fulfill()
+            dispatch_async(Queue.Main.queue, {
+                guard let expectation = expectation else { print("Test: \(#function): Finished expectation after timeout"); return }
+                expectation.fulfill()
+            })
```

Comment:
In many cases, agreed.

In cases like the one above, which is testing completion blocks, I think it may unfortunately need to remain in the completion block, due to the difference between completion blocks and DidFinishObservers.